spec/linux/kernel.go

```diff
@@ -34,9 +33,15 @@ func (g *KernelGenerator) Generate() (interface{}, error) {
 			kernelLogger.Errorf("Failed to run %s %s (skip this spec): %s", command[0], command[1], err)
 			return nil, err
 		}
-		str := strings.TrimSpace(string(out))
-
-		results[key] = str
+		results[key] = strings.TrimSpace(string(out))
+	}
+	out, err := exec.Command("uname", "-o").Output()
+	if err == nil {
+		results["os"] = strings.TrimSpace(string(out))
+	} else {
+		// In some environments like busybox there is no `uname -o` option,
```

Comment:
[nit]
Recent busybox itself implements `uname -o`, therefore this comment should be "*old* busybox"